### PR TITLE
Fixed Android's Border background to use a Transparent bg if not othe…

### LINKED ISSUE #16547: Border background is wrong by default on Android and text cannot be seen (it's correct on Windows)
### Description

When I wrap a Border around my content, the default background color is black on Android and I can't see Label text. Is the Border really supposed to be filled black on Android? I thought it should be transparent by default.

### Android:

* image

### Windows:

* image

### Steps to Reproduce

I was using **.NET 8 preview 7**.

1. In Visual Studio, File|New|Blank MAUI app
2. Add a Border around the default ScrollView in MainPage.xaml:
    * image
3. Change debug target to Android emulator
    * image
4. F5 to build and debug

### Link to public reproduction project repository

_No response_

### Version with bug

Unknown/Other

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

net8.0-android21.0

### Did you find any workaround?

My workaround is to manually set the background to transparent:

* image

### Relevant log output

_No response_

### LINKED ISSUE #17506: Border bugs entries
### Description

If you create a border and then inside you have an entry to put stroke and color. No matter what you do the whole border and entry will get a dark background even in Light theme.

### Steps to Reproduce

1. Create a File > New .NET MAUI App
2. Add a Border with an Entry inside: `<Border><Entry/> </Border>`
3. Run the application and see the bug.

### Link to public reproduction project repository

_No response_

### Version with bug

8.0.0-rc.1.9171

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

8.0.0-preview.4.8333

### Affected platforms

Android

### Affected platform versions

Android 13, Android 12

### Did you find any workaround?

No workaround for now.

### Relevant log output

_No response_

### LINKED ISSUE #17547: [Android] Border default background is black
### Description

Without setting the background property the Border set a black background:

image

In this case, something like a default Label is not visible. In .NET 7 the default value was a Transparent background.

### Steps to Reproduce

1. Take a look to the attached screenshot and create a Border without Background.
2. Launch the sample on Android.

### Link to public reproduction project repository

_No response_

### Version with bug

8.0.0-rc.1.9171

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

_No response_

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by jstedfast**
### Description of Change

Fixed Android's Border implementation to use a transparent bg when painting the border if a bg color is left unspecified.

### Issues Fixed

Fixes #17547
Fixes #17506
Fixes https://github.com/dotnet/maui/issues/16547

### CODE REVIEW

**Review by jknaudt21 — DISMISSED** [review]

**Comment by PureWeen** [comment]
Can we add a test?

**Comment by jstedfast** [comment]
@PureWeen added.

**Review by jsuarezruiz — DISMISSED** [review]

File: src/Core/src/Graphics/MauiDrawable.Android.cs [hunk 1]
```diff
@@ -522,10 +522,13 @@ void SetBackground(APaint platformPaint)
 					platformPaint.Color = _backgroundColor.Value;
 #pragma warning restore CA1416
 				}
+				else if (_background != null)
```
Comment (jfversluis) [inline src/Core/src/Graphics/MauiDrawable.Android.cs:525]: Nit

```suggestion
				else if (_background is not null)
```

**Review by jfversluis — DISMISSED** [review]

**Review by PureWeen — DISMISSED** [review]
Related https://github.com/dotnet/maui/issues/17631

**Comment by BretJohnson** [comment]
FYI - I also added a visual test (Appium test with screenshot) for this scenario, as part of: https://github.com/dotnet/maui/pull/17633. I added that partially to better test on iOS (as iOS currently fails here, for reasons unrelated to this PR which only has Android changes).

**Review by BretJohnson — APPROVED** [review]

**Review by jfversluis — APPROVED** [review]

**Pull request merged**